bin/cli.mjs

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env node
+
+import { Command } from 'commander';
+import { glob } from 'glob';
+import { basename, join } from 'node:path';
+import { argv, exit } from 'node:process';
+
+/**
+ * @fileoverview
+ *
+ * The CLI used to help create API documentation for the Node.js project.
+ * The CLI processes files and generates API documentation based on the provided options.
+ */
+
+/**
+ * Process files based on provided command line arguments.
+ *
+ * @param {string} filePath - The path of the file to be processed.
+ * @param {Options['output']} outputDir - The directory where the processed file will be saved.
+ * @param {Options['target']} target - The processing mode to be applied to the file.
+ * @returns {Promise<void>} A promise that resolves when the file processing is complete.
+ */
+const processFile = async (filePath, outputDir, target) => {
+  try {
+    const fileName = basename(filePath);
+    const outputPath = join(outputDir, fileName);
+
+    console.info(
+      `Processing file ${filePath} to ${outputPath} with target ${target}...`
+    );
+
+    // TODO: Implement file processing logic here
+    return new Promise();
+  } catch (error) {
+    console.error(`Error processing file ${filePath}:`, error);
```

Comment:
We could use [`util.styletext`](https://nodejs.org/docs/v20.15.1/api/util.html#utilstyletextformat-text) to get cooler logs.

What do you think ?

Reply:
I think it's a good idea, but I prefer to open this PR as a different PR to avoid making it too large 👀 

As a follow-up, I can open a PR for logging. It would be a simple helper method under util that would detail and color the outputs based on the log level we provide from the CLI/environment, and add time, trace, etc. information if necessary

Reply:
Cool! Let's do this.

Reply:
I will mark this as resolved and start working, thank you for the suggestion @AugustinMauroy